cmd/namespace_registry.go

```diff
@@ -0,0 +1,253 @@
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/x509"
+	"crypto/elliptic"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"math/big"
+	"net/http"
+	"bytes"
+	"bufio"
+)
+
+type JWKS struct {
+	Keys []JWK `json:"keys"`
+}
+
+type JWK struct {
+	Crv string `json:"crv"`
+	Kid string `json:"kid"`
+	Kty string `json:"kty"`
+	X   string `json:"x"`
+	Y   string `json:"y"`
+}
+
+func loadPrivateKey(privateKeyPath string) (*ecdsa.PrivateKey, error) {
+	keyInBytes, err := ioutil.ReadFile(privateKeyPath)
+	if err != nil {
+		return nil, err
+	}
+	block, _ := pem.Decode(keyInBytes)
+	privateKey, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	return privateKey.(*ecdsa.PrivateKey), nil
+}
+
+func loadPublicKey(publicKeyPath string) (*ecdsa.PublicKey, error) {
+	keyInBytes, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+	var jwks JWKS
+	err = json.Unmarshal(keyInBytes, &jwks)
+	if err != nil {
+		return nil, err
+	}
+	keyData := jwks.Keys[0]  // Assumes there's at least one key
```

Comment:
Similar to the private key comment, I think that we should use the loader here: https://github.com/PelicanPlatform/pelican/blob/main/config/init_server_creds.go#L204

In general, let's aim to use the JWT library instead of rolling our own.

Reply:
✅